src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -945,6 +974,68 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                     dtγ * ᶠtridiagonal_matrix_c3 ⋅
                     DiagonalMatrixRow(adjoint(CT3(Y.f.sgsʲs.:(1).u₃))) - (I_u₃,)
             end
+            # add updraft mass flux contributions to grid-mean
+            if use_derivative(sgs_mf_flag)
+
+                (; ᶜgradᵥ_ᶠΦ, ᶜρʲs, ᶠu³ʲs, ᶜtsʲs) = p
+                (; bdmr_l, bdmr_r, bdmr) = p
+                is_third_order = edmfx_upwinding == Val(:third_order)
+                ᶠupwind = is_third_order ? ᶠupwind3 : ᶠupwind1
+                ᶠset_upwind_bcs = Operators.SetBoundaryOperator(;
+                    top = Operators.SetValue(zero(CT3{FT})),
+                    bottom = Operators.SetValue(zero(CT3{FT})),
+                ) # Need to wrap ᶠupwind in this for well-defined boundaries.
+                UpwindMatrixRowType =
+                    is_third_order ? QuaddiagonalMatrixRow : BidiagonalMatrixRow
+                ᶠupwind_matrix =
+                    is_third_order ? ᶠupwind3_matrix : ᶠupwind1_matrix
+                ᶠset_upwind_matrix_bcs = Operators.SetBoundaryOperator(;
+                    top = Operators.SetValue(
+                        zero(UpwindMatrixRowType{CT3{FT}}),
+                    ),
+                    bottom = Operators.SetValue(
+                        zero(UpwindMatrixRowType{CT3{FT}}),
+                    ),
+                ) # Need to wrap ᶠupwind_matrix in this for well-defined boundaries.
+                ᶜkappa_mʲ = p.ᶜtemp_scalar
+                @. ᶜkappa_mʲ =
+                    TD.gas_constant_air(thermo_params, ᶜtsʲs.:(1)) /
+                    TD.cv_m(thermo_params, ᶜtsʲs.:(1))
+
+
+                # Placeholders for jacobian contributions of updraft massflux to grid-mean
+                # Initialize all derivatives to zero
+
+                d_flux_etot_d_htot = ᶠbidiagonal_matrix_ct3
```

Comment:
You don't need any of this but if you want to keep this for next PR it's ok.

Reply:
I'll leave in for now since this isn't on by default and these will be immediately used in the next PR